packages/react/src/components/Slider/Slider.tsx

```diff
@@ -0,0 +1,236 @@
+    inputRef: customPropTypes.ref,
+    max: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+    min: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+    onChange: PropTypes.func,
+    step: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+    vertical: PropTypes.bool,
+  }
+
+  static defaultProps: SliderProps = {
+    accessibility: sliderBehavior,
+    getA11yValueMessageOnChange: ({ value }) => String(value),
+    iconPosition: 'start',
+    max: 100,
+    min: 0,
+    step: 1,
+  }
+
+  static autoControlledProps = ['value']
+
+  getInitialAutoControlledState(): Partial<SliderState> {
+    return { value: '50' }
+  }
+
+  handleChange = (e: React.ChangeEvent) => {
+    const value = _.get(e, 'target.value')
+    _.invoke(this.props, 'onChange', e, { ...this.props, value })
+    this.trySetState({ value })
+  }
+
+  handleFocus = (e: React.FocusEvent) => {
+    this.setState({ isFromKeyboard: isFromKeyboard() })
+    _.invoke(this.props, 'onFocus', e, this.props)
+  }
+
+  renderIcon = (variables: ComponentVariablesObject, styles: ComponentSlotStylesPrepared) => {
+    const { icon } = this.props
+
+    return Icon.create(icon, {
+      defaultProps: {
+        styles: styles.icon,
+        variables: variables.icon,
+      },
+    })
+  }
+
+  renderComponent({
+    ElementType,
+    classes,
+    accessibility,
+    variables,
+    styles,
+    unhandledProps,
+  }: RenderResultConfig<SliderProps>) {
+    const { iconPosition, input, inputRef } = this.props
+    const { value = '' } = this.state
+    const [htmlInputProps, restProps] = partitionHTMLProps(unhandledProps)
+    const type = 'range'
+
+    return (
+      <ElementType
+        className={classes.root}
+        {...accessibility.attributes.root}
+        {...restProps}
+        {...applyAccessibilityKeyHandlers(accessibility.keyHandlers.root, unhandledProps)}
+      >
+        {iconPosition === 'start' && this.renderIcon(variables, styles)}
+        <div className={classes.inputWrapper}>
+          <Ref
+            innerRef={(inputElement: HTMLElement) => {
+              handleRef(this.inputRef, inputElement)
+              handleRef(inputRef, inputElement)
+            }}
+          >
+            {Box.create(input || type, {
+              defaultProps: {
+                ...htmlInputProps,
+                ...accessibility.attributes.input,
+                className: Slider.slotClassNames.input,
+                as: 'input',
+                type,
+                value,
+                onChange: this.handleChange,
+                onFocus: this.handleFocus,
+                styles: styles.input,
+                ...applyAccessibilityKeyHandlers(accessibility.keyHandlers.input, htmlInputProps),
+              },
+            })}
+          </Ref>
+        </div>
+        {iconPosition !== 'start' && this.renderIcon(variables, styles)}
```

Comment:
Should the icon optionally be a button instead of just icon since it needs the ability to perform some action when clicked? when clicked it should mute
![image](https://user-images.githubusercontent.com/828692/60930166-3cfd2680-a269-11e9-89c7-6fc174551124.png)

Reply:
We will also need hover/focus styles for the icon button

Reply:
good point, let me think about it; will update the thread soon
@codepretty is the 20 volume mark part of the slider? I don't see that in the design files I have

Reply:
discussed with the team and we decided not to do that in v1 since it's not a requirement that's generic enough; we might reconsider; I will add an example on how to compose the `Slider` with `Button` component to achieve the behavior you're mentioning:
> it needs the ability to perform some action when clicked

Reply:
That screenshot is from the Windows default speaker controls. I was showing that to show that its a common thing. But more likely that we should have a Slider usage example that is for volume control? Or a separate component altogether? The icon should probably be mapped to parts of the range since the icon itself updates at different places on the slider to indicate volume level.

![image](https://user-images.githubusercontent.com/828692/60993746-e85ab980-a303-11e9-9205-fb29be2d2ca4.png)

Reply:
sure, np @codepretty 
I just added an example to show how to compose the Slider with Button component to achieve the behavior you're mentioning:

> it needs the ability to perform some action when clicked

so I'm resolving this comment